### PR TITLE
[T04]: Search job by title

### PR DESCRIPTION
**Pull request by vinhvinhvinh**
Stories:
Link code feature: https://github.com/fabatek/onboarding-job-board-frontend/tree/feature/vinhnt/t04_search_job_by_title
Link deploy surge: http://job_board_vinhnt.surge.sh/
Link API to see specific job totals: https://623804ad0a54d2ceab6f951b.mockapi.io/api/jobs

Notes:

1. File Modified
- under_header ( handle form )

- MainContent.tsx 
	- remove interface IsLoading, commuted jobState.loading(redux Store) for loading(hooks) 
	- remove loading props
- Home/index.tsx ( remove loading State )

- jobs-action.ts 
- add getJobsFilter function

job-reducer.ts
- add the "keyword" into IState
- add another action named JOBS_FILTER

index.test.ts
- add describe GET /job 

2. Deploy Update
Result:
image

### CODE REVIEW

File: src/components/Header/UnderHeader/under_header.tsx [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
-import { useSelector } from "react-redux";
+import React, { useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { getJobsFilter } from "../../../redux/jobs/jobs-action";
 import { RootStore } from "../../../redux/store";
 
 const UnderHeader: React.FC = () => {
+  const dispatch = useDispatch();
   const jobState = useSelector((state: RootStore) => state.job);
+  const [inputSearch, SetInputSearch] = useState<string>("");
```
Comment (ducdmfaba) [inline src/components/Header/UnderHeader/under_header.tsx:9]: Why did you upper first character of `SetInputSearch`? It should be `setInputSearch`.

File: src/components/Main/MainContent.tsx [hunk 2]
```diff
@@ -29,11 +24,16 @@ const MainContent = ({ loading }: IsLoading): ReactElement => {
...
+
   const JobMain = ({ currentItems }: Items): ReactElement => {
     return (
       <>
         <h2>Top Jobs</h2>
-        {loading ? (
+        {jobState.loading ? (
           <div className="text-center">
             <div className="spinner-border" role="status">
               <span className="sr-only"></span>
```
Comment (ducdmfaba) [inline src/components/Main/MainContent.tsx:39]: Please create `<Loading />` component to reuse.

File: src/redux/jobs/jobs-action.ts [hunk 3]
```diff
@@ -23,4 +25,27 @@ const getJobs = () => async (dispatch:Dispatch<JobDispatchTypes>):Promise<void>=
...
+        });
+    }catch(err){
+        dispatch({
+            type:JOBS_FAIL,
+        });
+    }
+};
+
+
+export {getJobs, getJobsFilter};
```
Comment (ducdmfaba) [inline src/redux/jobs/jobs-action.ts:51]: The format codes look incorrect. Please format it again.

File: src/redux/jobs/jobs-action.ts [hunk 4]
```diff
@@ -23,4 +25,27 @@ const getJobs = () => async (dispatch:Dispatch<JobDispatchTypes>):Promise<void>=
...
+            type:JOBS_LOADING
+        });
+
+        const response=await axios.get(`${JOBS_URL}?name=${searchKey}`);
+
+        dispatch({
+            type:JOBS_FILTER,
+            payload:{
+                jobs:response.data,
+                keyword:searchKey,
```
Comment (ducdmfaba) [inline src/redux/jobs/jobs-action.ts:40]: Do we need to store this?

**Review by ducdmfaba — CHANGES_REQUESTED** [review]

**Comment by ducdmfaba** [comment]
image

The grid shows incorrectly data

**Comment by ducdmfaba** [comment]
image

The endpoint without filter is incorrect

File: src/components/Header/UnderHeader/under_header.tsx (hunk 1 above)
Comment (vinhvinhvinh) [reply src/components/Header/UnderHeader/under_header.tsx:9]: sorry Sir, my fatal mistake

File: src/redux/jobs/jobs-action.ts (hunk 4 above)
Comment (vinhvinhvinh) [reply src/redux/jobs/jobs-action.ts:40]: thank u Sir! i got this

**Review by ducdmfaba — APPROVED** [review]

**Pull request merged**